storage/postgres/storage.go

```diff
@@ -229,7 +235,9 @@ func (ps *Storage) Delete(ctx context.Context, objType types.ObjectType, criteri
 	if err != nil {
 		return nil, err
 	}
-	rows, err := deleteAllByFieldCriteria(ctx, ps.pgDB, entity.TableName(), entity, criteria)
+
+	qBuilder := NewQueryBuilder(ps.pgDB)
+	rows, err := qBuilder.WithCriteria(criteria...).Return("*").Delete(ctx, entity)
```

Comment:
This order is strange. It seems more natural to say `Delete(ctx, entity).WithCriteria().Return()`

Reply:
If it is implemented the way you propose, it will need to have one more method call to signify execution of query. For example: `qBuilder.Delete().WithCriteria().Return().Result()`